File: 1º Ano/2º Semestre/Programação Orientada a Objetos/Labs/Lab02/Room.java
public class Room
{
    private int number;
    private String name;
    private double area;
   
    public Room(String name, double area){
         this.name = name;
         this.area = area;
         this.number = this.number + 1;
    }
    
    public Room(){
        this.name = "";
        this.area = 0.00;
        this.number = this.number + 1;
    }
    
    public int getNumber(){
        return this.number;
    }
    
    public String getName(){
        return this.name;
    }
    
    public double getArea(){
        return this.area;
    }
    
    public void setName(String name){
        if (name != "") {
             this.name = name;
        }
    }
    
    public void setArea(double area){
        if (area != 0.00) {
            this.area = area;
        }
        
    }
    
    public String toString(){
        return "Room Name: " + this.name;
    }
       
}

File: 2º Ano/2º Semestre/Computação Distribuida/Projectos/tp2-team-alexandrecoelho-sergioverissimo-t2/room.py
class Room:
    def __init__(self, name):
        self.name = name
        self.active_users = []

    def add_user(self, user):
        self.active_users.append(user)

    def add_message(self, message):
        self.messages.append(message)

    def user_exist(self, name):
        for user in self.active_users:
            if user.get_name() == name:
                return True
        return False

    def get_name(self):
        return self.name

    def get_active_users(self):
        return self.active_users

    def remove_user(self, user):
        self.active_users.remove(user)

    def broadcast(self, msg):
        for user in self.active_users:
            user.add_received_room_message(msg)

    def to_string(self):
        ret = 'Room: ['
        ret += 'Name: ' + self.name + ', Active Users: ['
        for user in self.active_users:
            ret += user.to_string() + ', '
        ret += ']]'
        return ret
